pkg/core/managers/apis/mesh/mesh_manager.go

```diff
@@ -89,6 +91,10 @@ func (m *meshManager) Delete(ctx context.Context, resource core_model.Resource,
 	if err := m.builtinCaManager.Delete(ctx, name); err != nil {
 		return errors.Wrapf(err, "failed to delete Builtin CA for a given mesh")
 	}
+	// delete associated resources
+	if err := m.store.DeleteMany(ctx, core_store.DeleteManyByMesh(opts.Name)); err != nil {
```

Comment:
I would expect the logic that iterates over all supported resource types in here (rather than in `k8s` store)